File: 2026/07-July/16-Thursday/solution.java
import java.util.Arrays;

class Solution {

    public long gcdSum(int[] nums) {
        int n = nums.length;
        int[] prefix = new int[n];

        int currentMax = 0;

        for (int i = 0; i < n; i++) {
            currentMax = Math.max(currentMax, nums[i]);
            prefix[i] = gcd(nums[i], currentMax);
        }

        Arrays.sort(prefix);

        long answer = 0;

        int left = 0;
        int right = n - 1;

        while (left < right) {
            answer += gcd(prefix[left], prefix[right]);
            left++;
            right--;
        }

        return answer;
    }

    private int gcd(int a, int b) {
        while (b != 0) {
            int temp = a % b;
            a = b;
            b = temp;
        }
        return a;
    }
}

File: 2026/07-July/16-Thursday/solution.py
from math import gcd
from typing import List

class Solution:
    def gcdSum(self, nums: List[int]) -> int:
        prefix = []
        current_max = 0

        for value in nums:
            if value > current_max:
                current_max = value
            prefix.append(gcd(value, current_max))

        prefix.sort()

        answer = 0
        left = 0
        right = len(prefix) - 1

        while left < right:
            answer += gcd(prefix[left], prefix[right])
            left += 1
            right -= 1

        return answer
